inc/blocks/namespace.php

```diff
@@ -116,8 +161,13 @@ function on_save_post( int $post_ID, WP_Post $post, bool $update ) : void {
 		] );
 		$posts = array_values( $posts ); // Reset keys.
 
-		// Generate a default using the current post title and instance number in the content.
-		$default_title = sprintf( '%s (XB %s)', $post->post_title, $index + 1 );
+		if ( $post_title ) {
```

Comment:
If we pass the context and a default title we can probably make this logic a bit more generic.